pkg/resources/statefulsets/statefulset.go

```diff
@@ -276,56 +279,148 @@ func createResources(r *corev1.ResourceRequirements) corev1.ResourceRequirements
+			SuccessThreshold:    successThreshold,
+			FailureThreshold:    failureThreshold,
 		}
 	}
 	return &corev1.Probe{
 		Handler: corev1.Handler{
 			HTTPGet: &corev1.HTTPGetAction{
-				Path: "/health/live",
+				Path: "/health/ready",
 				Port: intstr.FromString("admin"),
 			},
 		},
-		InitialDelaySeconds: 60,
+		InitialDelaySeconds: initialDelaySeconds,
+		TimeoutSeconds:      timeoutSeconds,
+		PeriodSeconds:       periodSeconds,
+		SuccessThreshold:    successThreshold,
+		FailureThreshold:    failureThreshold,
 	}
 }
 
-// createReadinessProbe create an Exec probe if the SERVER_READINESS_SCRIPT env var is present
+// createStartupProbe create an Exec probe if the SERVER_LIVENESS_SCRIPT env var is present
 // *and* the application is not using Bootable Jar.
+// Otherwise, it creates a HTTPGet probe that checks the /health/live endpoint on the admin port.
 //
-// If defined, the SERVER_READINESS_SCRIPT env var must be the path of a shell script that
+// If defined, the SERVER_LIVENESS_SCRIPT env var must be the path of a shell script that
 // complies to the Kubernetes probes requirements.
-func createReadinessProbe(w *wildflyv1alpha1.WildFlyServer) *corev1.Probe {
-	readinessProbeScript, defined := os.LookupEnv("SERVER_READINESS_SCRIPT")
-	if defined && !w.Spec.BootableJar {
+// If SERVER_LIVENESS_SCRIPT script does not exist, then the Probe will execute the script defined by SERVER_LIVENESS_SCRIPT_FALLBACK
+// If this script does not exist, the probe will execute and http get by using curl.
+func createStartupProbe(w *wildflyv1alpha1.WildFlyServer) *corev1.Probe {
+	livenessProbeScript, defined := os.LookupEnv("SERVER_LIVENESS_SCRIPT")
+
+	var initialDelaySeconds int32 = 60
+	var timeoutSeconds int32
+	var periodSeconds int32
+	var successThreshold int32
+	var failureThreshold int32
+
+	if w.Spec.StartupProbe != nil {
+		initialDelaySeconds = w.Spec.StartupProbe.InitialDelaySeconds
+		timeoutSeconds = w.Spec.StartupProbe.TimeoutSeconds
+		periodSeconds = w.Spec.StartupProbe.PeriodSeconds
+		successThreshold = w.Spec.StartupProbe.SuccessThreshold
+		failureThreshold = w.Spec.StartupProbe.FailureThreshold
+
+		//Only if StartupProbe was defined by the user
+		return createLivenessStartupCommonProbe(defined, w.Spec.BootableJar, livenessProbeScript, initialDelaySeconds, timeoutSeconds, periodSeconds, successThreshold, failureThreshold)
+	}
+
+	return nil
+}
+
+// creates the common parts for the Linevess and Startup Probes
+func createLivenessStartupCommonProbe(scriptDefined bool, bootableJar bool, livenessProbeScript string, initialDelaySeconds int32, timeoutSeconds int32, periodSeconds int32, successThreshold int32, failureThreshold int32) *corev1.Probe {
+	if scriptDefined && !bootableJar {
+		livenessProbeScriptFallback, definedFallback := os.LookupEnv("SERVER_LIVENESS_SCRIPT_FALLBACK")
+		if !definedFallback {
+			livenessProbeScriptFallback = "curl 127.0.0.1:" + strconv.Itoa(int(resources.HTTPManagementPort)) + "/health/live"
```

Comment:
This requires that the management API is bound to 0.0.0.0 or to loopback directly, right?

Reply:
Yes, that's correct, if the management interface is not bound to 0.0.0.0 or to loopback, the probe will fail